nixos/modules/services/databases/redis.nix

```diff
@@ -72,7 +72,27 @@ in {
               defaultText = literalExpression ''
                 if name == "" then "redis" else "redis-''${name}"
               '';
-              description = "The username and groupname for redis-server.";
+              description = ''
+                User account under which this instance of redis-server runs.
+
+                ::: {.note}
+                If left as the default value this user will automatically be
+                created on system activation, otherwise you are responsible for
+                ensuring the user exists before the redis service starts.
+              '';
+            };
+
+            group = mkOption {
+              type = types.str;
+              default = config.user;
+              description = ''
+                Group account under which this instance of redis-server runs.
+
+                ::: {.note}
+                If left as the default value this group will automatically be
+                created on system activation, otherwise you are responsible for
+                ensuring the group exists before the redis service starts.
+              '';
```

Comment:
I'm uncertain what will happen if you specify a group that the user isn't a member of, and whether that's something we should assert or automatically set up.